README.md

```diff
@@ -214,6 +214,75 @@ client := sdk.NewClientWithOpts(
 )
 ```
 
+## Build functions
+
+Use the OpenFaaS [OpenFaaS Function Builder API](https://docs.openfaas.com/openfaas-pro/builder/) to build functions from code.
+
+The Function Builder API provides a simple REST API to create your functions from source code. The API accepts a tar archive with the function build context and build configuration. The SDk provides methods to create this tar archive and invoke the build API.
+
+If you functions are using a language template you will need to make sure the required templates are available on the file system. How this is done is up to your implementation. Templates can be pulled from a git repository, copied from an S3 bucket, downloaded with an http call or fetched with the faas-cli.
+
+```go
+functionName := "hello-world"
+handler := "./hello-wold"
+lang := "node18"
+
+// Get the HMAC secret used for payload authentication with the builder API.
+payloadSecret, err := os.ReadFile("payload.txt")
+if err != nil {
+	log.Fatal(err)
+}
+payloadSecret := bytes.TrimSpace(payloadSecret)
+
+// Initialize a new builder client.
+builderURL, _ := url.Parse("https://builder.example.dev")
```

Comment:
This should use a cluster local address. We do not want to encourage people to expose the builder api on the Internet.

http://.... etc